src/main/rx/spektrum.c

```diff
@@ -157,6 +158,147 @@ static int8_t dBm2range (int8_t dBm)
 }
 #endif
 
+ #ifdef USE_SPEKTRUM_VTX_CONTROL
+ #ifdef VTX_COMMON
+
+// Use the common set/get-frequncy API.
+// Much more generic than "knowing" someone elses indexes and trying to convert..
+#define USE_VTX_COMMON_FREQ_API
+
+#ifdef USE_VTX_COMMON_FREQ_API
+    const uint16_t SpektrumVtxfrequencyTable[SPEKTRUM_VTX_BAND_COUNT][SPEKTRUM_VTX_CHAN_COUNT] =
+      {
+        { 5740, 5760, 5780, 5800, 5820, 5840, 5860, 5880 }, // FatShark
```

Comment:
Does it make sense to redefine the frequency table here? We should probably just delete it from `vtx_string.c` and make everybody use this. (Or the other way round.)

Reply:
This is the SPEKTRUM frequency table. The frequencies indexed by SPEKTRUM band and channel id's. Not necessarily the same as defined by some other parties.

Reply:
To me it looks like these are the same frequencies that are used by manufacturers other than Spektrum as well. Why do they claim it to be 'their' frequency table?

Reply:
Different order, different indexing. This table is only used to lookup frequency using Spektrum indexes. Making us independent on the diffs in indexes and order. But the table will be disabled when turning of USE_VTX_COMMON_FREQ_API.  Instead a band translation table will be used, and channel index bumping. Nice hugh....

Reply:
Seems a bit silly to add yet another vendor specific data structure, instead of making this driver use a shared data structure. Proprietary protocols should be based on generic implementations as much as possible, and not reimplement them.

Reply:
That was the whole idea of using a generic frequency parameter, not brand specific indexes. Isolating each brand in its own modules, spektrum specific tables in the spektrum driver. I only wish the VTX devices would do the same, via the vtx_common middle ware layer. But unfortunatlly not. Now we have to have all kinds of VTX brand specific stuff in the spektrum modules. @null has offered to look this over and add some abstraction.

Reply:
I still don't get where you see 'brand specific indexes' for band / channel selection. Can you please point that out?

Reply:
Maybe not "brands" when it comes to bands/channels. Still different order, 0 based indexes vs 1 based ( and 0 based in some places).  Power settings are by brand. All needing conversion. 
Only thing everyone seems to agree on is pitmode ON/OFF. 

https://github.com/SpektrumFPV/SpektrumDocumentation/blob/master/Telemetry/Spektrum%20VTX%20Interfacing.pdf : 

```
BAND:
0 = FatShark
1 = Raceband
2 = E
3 = B
4 = A
5-7 = Reserved 
CHAN (Channel): 
   0    1    2    3    4    5    6    7
5740 5760 5780 5800 5820 5840 5860 5880 FatShark
5658 5695 5732 5769 5806 5843 5880 5917 Raceband
5705 5685 5665 5645 5885 5905 5925 5945 Band E
5733 5752 5771 5790 5809 5828 5847 5866 Band B
5865 5845 5825 5805 5785 5765 5745 5725 Band A
```

src/main/io/vtx_string.c : 

```
const uint16_t vtx58frequencyTable[VTX_STRING_BAND_COUNT][VTX_STRING_CHAN_COUNT] =
{
    { 5865, 5845, 5825, 5805, 5785, 5765, 5745, 5725 }, // Boscam A
    { 5733, 5752, 5771, 5790, 5809, 5828, 5847, 5866 }, // Boscam B
    { 5705, 5685, 5665, 5645, 5885, 5905, 5925, 5945 }, // Boscam E
    { 5740, 5760, 5780, 5800, 5820, 5840, 5860, 5880 }, // FatShark
    { 5658, 5695, 5732, 5769, 5806, 5843, 5880, 5917 }, // RaceBand
};

const char * const vtx58BandNames[] = {
    "--------",
    "BOSCAM A",
    "BOSCAM B",
    "BOSCAM E",
    "FATSHARK",
    "RACEBAND",
};
```
Also PR #4265